homeassistant/components/rointe/rointe_entity.py

```diff
@@ -0,0 +1,105 @@
+"""Rointe devices entity model."""
+from __future__ import annotations
+
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import (
+    ATTRIBUTION,
+    DOMAIN,
+    ROINTE_HA_SIGNAL_UPDATE_ENTITY,
+    ROINTE_MANUFACTURER,
+)
+from .coordinator import RointeDataUpdateCoordinator
+from .device_manager import RointeDevice, RointeDeviceManager
+
+
+class RointeHAEntity(CoordinatorEntity):
+    """Rointe entity base class."""
+
+    def __init__(self, coordinator: RointeDataUpdateCoordinator, name, unique_id):
+        """Initialize the entity."""
+        super().__init__(coordinator)
+        self._unique_id = f"rointe-{unique_id}"
+        self._name = name
+
+    def get_device_manager(self) -> RointeDeviceManager:
+        """Get the device manager."""
+        return self.coordinator.device_manager
+
+    @property
+    def unique_id(self):
+        """Return the unique id."""
+        return self._unique_id
```

Comment:
If you store the unique ID in `_attr_unique_id` instead of `_unique_id`, you can avoid redefining this property.